integration_tests/test_str_01.py

```diff
@@ -49,6 +49,12 @@ def test_constant_str_subscript():
     assert "abc"[2] == "c"
     assert "abc"[:2] == "ab"
 
+def test_str_slice_step():
+    s: str
+    s = "abcdefghijk"
+    assert s[3:12:3] == "dgj"
+    assert s[12:3:-3] == "mjg"
```

Comment:
Can you please add more tests? Like these:

```python
>>> s = "abcdefg"
>>> s[::]
'abcdefg'
>>> s[1:4:]
'bcd'
>>> s[1::-2]
'b'
>>> s[:4:5]
'a'
>>>
```

Reply:
In addition to the above also try passing non-constant expressions to the inputs. Say like this,

```python
start: i32
end: i32
step: i32
s = "abcdefg"
start = 1
end = 4
step = 1
assert s[start:end:step] == "bcd"
assert s[start:end - 1:step] == "bc"
```

Reply:
I have added more tests now !